### PR TITLE
Preview dbt Cloud feature: Slim CI

### PR DESCRIPTION
**Pull request by jtcohen6**
## Description & motivation

- Document job deferral in dbt Cloud as a preview feature
- Update best practice on slim CI

@drewbanin Could you take a look at the language here? I don't think we have any other "preview" features in dbt Cloud that are available on all plans.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

File: website/docs/docs/guides/best-practices.md [hunk 1]
```diff
@@ -111,16 +111,20 @@ To merge code changes with confidence, you want to know that those changes will
...
 ```bash
-dbt run -m state:modified --defer --state path/to/prod/artifacts
-dbt test -m state:modified
+dbt run -m state:modified+ --defer --state path/to/prod/artifacts
+dbt test -m state:modified+
 ```
 
+:::info [β] Beta Feature
+This is net-new functionality in v0.18.0, with iterative improvements to come.
+You may need to adjust syntax for your project project. If you encounter unexpected
```
Comment (clrcrl) [inline website/docs/docs/guides/best-practices.md:124]:
```suggestion
You may need to adjust syntax for your project. If you encounter unexpected 
```

Comment (clrcrl) [inline website/docs/docs/guides/best-practices.md]:
```suggestion
You may need to adjust syntax for your project. If you encounter unexpected 
```

**Review by clrcrl — APPROVED** [review]
approved pending minor correction

File: website/docs/docs/guides/best-practices.md (hunk 1 above)
Comment (jtcohen6) [reply website/docs/docs/guides/best-practices.md]: ay nice catch ty

**Comment by jtcohen6** [comment]
Thanks for the review @clrcrl! I added one more tiny note about both jobs needing to run dbt >= v0.18.0, given feedback from beta tester extraordinaire @aescay.

I'm going to merge this in so that I can announce it in dbt Slack. We can always adjust the language later.

**Comment by drewbanin** [comment]
Meeep - apologies for the delay. Glad you merged this one without me!

**Pull request merged**